### ISSUE TITLE
Use symlinks instead of copying config files.

### ISSUE THREAD

**Issue by mitchellallison** [Tedworthy/ConsistentConfig#1]
One of the caveats of the approach (as noted) is that you need to edit the files in the repo, not the files in the actual home directory.

Instead of copying files, this could be remedied by creating symbolic links to the files in the repo.

Of course, the caveat still remains that the files aren't necessarily committed or pushed back to GitHub.

**Comment by Tedworthy** [comment]
Yeah I like that approach, I guess there are cases for doing both.

The symlink idea is good because it means you can effectively edit either the "file" in your home directory or the file in the repo, as both point to the same thing.

However if you were to reorganise your files and move the directory (or even delete it) containing the repo, the symlink would no longer be valid - it means you're fully dependent the other directory containing the repo. The existing approach is therefore better in this case, as there is absolutely no dependency.

I would suggest that both options could be made available in the makefile, and you call a different make rule for each approach - `make copy` for the existing, and `make symlink` for the symlink idea. Covers both bases and opinions then, and would be easy to switch between both ideas if you so desired, just by calling another make command.

Another idea that would be worth weighing up is cloning this repo directly into your home directory. This would mean you'd have a ~/.git directory, and your currently used config files would actually be the files managed by the repo. That would break the bashrc/bash_profile issue in OS X, but actually that could be alleviated with a simple symlink from .bash_profile to .bashrc. Not sure how config files that normally reside in directories would work, such as something like ~/my_program/programconfig. If the repo path matched the one in your home directory I guess it would work fine - cloning would just affect the file in the repo. You'd need some form of "gitinclude", i.e. the exact opposite of gitignore to only include the config files in the repo, but as per this [SO question](http://stackoverflow.com/questions/1279533/is-there-a-way-to-tell-git-to-only-include-certain-files-instead-of-ignoring-cer) it appears that is easily possible.

Food for thought...
